### PR TITLE
docs(connectors): adds tasks to pause or stop connectors with state property configuration

### PR DESCRIPTION
**Pull request by PaulRMellor**
**Documentation**

Documents [Proposal 54](https://github.com/strimzi/proposals/blob/main/054-stopping-kafka-connect-connectors.md)

- Adds steps to use the new `state` config to stop or pause KC and MM2 connectors
- Moves Kafka Connect connector set up content up a level for easier navigation
- Moves KC and MM2 connector-related tasks to sit with config content for consistency
- Updates Kafka Connect and MM2 connector restart tasks to simplify and make consistent

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: documentation/modules/configuring/proc-manual-restart-mirrormaker2-connector-task.adoc [hunk 1]
```diff
@@ -2,45 +2,43 @@
 // assembly-management-tasks.adoc
 
 [id='proc-manual-restart-mirrormaker2-connector-task-{context}']
-= Performing restarts of MirrorMaker 2 connector tasks using annotations
+= Manually restarting MirrorMaker 2 connector tasks
 
-This procedure describes how to manually trigger a restart of a Kafka MirrorMaker 2 connector task by using a Kubernetes annotation.
+[role="_abstract"]
+Use the `restart-connector-task` annotation to manually trigger a restart of a MirrorMaker 2 connector.
```
Comment (scholzj) [inline documentation/modules/configuring/proc-manual-restart-mirrormaker2-connector-task.adoc:8]: Should we use the full annotation name?

```suggestion
Use the `strimzi.io/restart-connector-task` annotation to manually trigger a restart of a MirrorMaker 2 connector.
```

File: documentation/modules/configuring/proc-manual-restart-mirrormaker2-connector.adoc [hunk 2]
```diff
@@ -2,9 +2,10 @@
 // assembly-management-tasks.adoc
 
 [id='proc-manual-restart-mirrormaker2-connector-{context}']
-= Performing restarts of MirrorMaker 2 connectors using annotations
+= Manually restarting MirrorMaker 2 connectors
 
-This procedure describes how to manually trigger a restart of a Kafka MirrorMaker 2 connector by using a Kubernetes annotation.
+[role="_abstract"]
+Use the `restart-connector` annotation to manually trigger a restart of a MirrorMaker 2 connector.
```
Comment (scholzj) [inline documentation/modules/configuring/proc-manual-restart-mirrormaker2-connector.adoc:8]: Same as above ...

```suggestion
Use the `strimzi.io/restart-connector` annotation to manually trigger a restart of a MirrorMaker 2 connector.
```

File: documentation/modules/configuring/proc-manual-restart-mirrormaker2-connector-task.adoc [hunk 3]
```diff
@@ -2,45 +2,43 @@
...
+Task IDs are non-negative integers, starting from 0.
 
-. To restart the connector task, annotate the `KafkaMirrorMaker2` resource in Kubernetes.
-In this example, `kubectl annotate` restarts task _0_ of a connector named `+my-source->my-target.MirrorSourceConnector+`:
+. Use the name and ID to restart the connector task by annotating the `KafkaMirrorMaker2` resource in Kubernetes:
 +
 [source,shell,subs="+quotes"]
 ----
-kubectl annotate KafkaMirrorMaker2 _KAFKAMIRRORMAKER-2-NAME_ "strimzi.io/restart-connector-task=my-source->my-target.MirrorSourceConnector:0"
+kubectl annotate KafkaMirrorMaker2 <mirrormaker_cluster_name> "strimzi.io/restart-connector-task=<mirrormaker_connector_name>:0"
```
Comment (scholzj) [inline documentation/modules/configuring/proc-manual-restart-mirrormaker2-connector-task.adoc:36]: I wonder if having an example also with the actual name would be helpful given the name is long and complicated.

File: documentation/modules/configuring/proc-manual-restart-mirrormaker2-connector.adoc [hunk 4]
```diff
@@ -19,27 +20,23 @@ This procedure describes how to manually trigger a restart of a Kafka MirrorMake
...
 ----
 
-. To restart the connector, annotate the `KafkaMirrorMaker2` resource in Kubernetes.
-In this example, `kubectl annotate` restarts a connector named `+my-source->my-target.MirrorSourceConnector+`:
+. Restart the connector by annotating the `KafkaMirrorMaker2` resource in Kubernetes:
 +
 [source,shell,subs="+quotes"]
 ----
-kubectl annotate KafkaMirrorMaker2 _KAFKAMIRRORMAKER-2-NAME_ "strimzi.io/restart-connector=my-source->my-target.MirrorSourceConnector"
+kubectl annotate KafkaMirrorMaker2 <mirrormaker_cluster_name> "strimzi.io/restart-connector=<mirrormaker_connector_name>"
```
Comment (scholzj) [inline documentation/modules/configuring/proc-manual-restart-mirrormaker2-connector.adoc:34]: Same as above, I would maybe add also an example with a real connector name.

**Review by scholzj — APPROVED** [review]
Some nits ... LGTM otherwise.

**Review by ppatierno — APPROVED** [review]
LGTM (apart Jakub's comments)

**Review by mimaison — APPROVED** [review]
LGTM, thanks

File: documentation/modules/configuring/proc-manual-restart-mirrormaker2-connector-task.adoc (hunk 3 above)
Comment (PaulRMellor) [reply documentation/modules/configuring/proc-manual-restart-mirrormaker2-connector-task.adoc:36]: Yes, we can.

```
In this example, task `0` for connector `my-connector` in the `my-mirror-maker-2` cluster is restarted:
+
[source,shell,subs="+quotes"]
----
kubectl annotate KafkaMirrorMaker2 my-mirror-maker-2 "strimzi.io/restart-connector-task=my-connector:0"
```

**Pull request merged**